core/src/buffer.rs

```diff
@@ -37,6 +37,13 @@ use scale_info::{
 #[derive(Clone, Default, Eq, Hash, Ord, PartialEq, PartialOrd, Decode, Encode, TypeInfo)]
 pub struct LimitedVec<T, E, const N: usize>(Vec<T>, PhantomData<E>);
 
+impl<T, E, const M: usize, const N: usize> From<[T; M]> for LimitedVec<T, E, N> {
```

Comment:
I'm against of assertions in runtime for such a type, that's mainly used in **non-test** code. It's not worth adding potentially not checked to be unreachable by tests panic in the trait impl that's needed only for tests.  
If there's no other stable solution, then it seems better to close the PR and wait for stable.

Reply:
+1